### PR TITLE
Fix reorg issues with phases

### PR DESCRIPTION
**Pull request by ManfredKarrer**

### CODE REVIEW

File: core/src/main/java/bisq/core/dao/governance/period/CycleService.java [hunk 1]
```diff
@@ -96,6 +96,28 @@ public void onParseBlockChainComplete() {
     // API
     ///////////////////////////////////////////////////////////////////////////////////////////
 
+    public void addFirstCycle() {
+        daoStateService.getCycles().add(getFirstCycle());
+    }
+
+    public int getCycleIndex(Cycle cycle) {
+        return (cycle.getHeightOfFirstBlock() - genesisBlockHeight) / cycle.getDuration();
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/governance/period/CycleService.java:104]: This is old code, but wouldn't this return the wrong index if the cycle length has been modified at some point.

**Review by sqrrm — APPROVED** [review]
utACK

File: core/src/main/java/bisq/core/dao/governance/period/CycleService.java (hunk 1 above)
Comment (ManfredKarrer) [reply core/src/main/java/bisq/core/dao/governance/period/CycleService.java:104]: Would need to look closer to the caller. Might be that the caller has an old cycle and that cause issues. Anyway that reorg topic will require more testing and effort to cover all edge cases. So I leave that for upcoming PRs...
